### PR TITLE
Add template for Azure Firewall using Firewall Policy for rule defini…

### PR DESCRIPTION
**Pull request by gimotwanMSFT**
…tionwq

# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-azurefirewall-with-firewallpolicy-sandbox-linux/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Create Azure Firewall sandbox setup
+
+<IMG SRC="https://azurequickstartsservice.blob.core.windows.net/badges/101-azurefirewall-sandbox-linux/PublicLastTestDate.svg" />&nbsp;
```
Comment (bmoore-msft) [inline 101-azurefirewall-with-firewallpolicy-sandbox-linux/README.md:3]: need to update all of the links for the new sample

File: 101-azurefirewall-with-firewallpolicy-sandbox-linux/metadata.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+  "type": "QuickStart",
+  "environments": [
+    "AzureCloud"
+  ],
+  "itemDisplayName": "Create a sandbox setup of Azure Firewall using Firewall Policy with Linux VMs",
```
Comment (bmoore-msft) [inline 101-azurefirewall-with-firewallpolicy-sandbox-linux/metadata.json:7]: display name is too long - open the file in VS code to check schema issues

File: 101-azurefirewall-with-firewallpolicy-sandbox-linux/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+            ]
+          }
+        },
+        {
+          "apiVersion": "2019-06-01",
+          "type": "ruleGroups",
+          "name": "AppRG",
+          "location": "[parameters('location')]",
+          "dependsOn": [
+            "[concat('Microsoft.Network/firewallPolicies/', variables('firewallPolicyName'), '/ruleGroups/', 'NetworkRG')]",
```
Comment (bmoore-msft) [inline 101-azurefirewall-with-firewallpolicy-sandbox-linux/azuredeploy.json:452]: remove concat and use resourceId

Comment (gimotwanMSFT) [reply 101-azurefirewall-with-firewallpolicy-sandbox-linux/azuredeploy.json]: Updated. Thanks for the suggestion!

File: 101-azurefirewall-with-firewallpolicy-sandbox-linux/metadata.json (hunk 2 above)
Comment (gimotwanMSFT) [reply 101-azurefirewall-with-firewallpolicy-sandbox-linux/metadata.json]: Fixed

File: 101-azurefirewall-with-firewallpolicy-sandbox-linux/README.md (hunk 1 above)
Comment (gimotwanMSFT) [reply 101-azurefirewall-with-firewallpolicy-sandbox-linux/README.md]: My bad. Thanks for pointing this out. Updated now.

**Comment by gimotwanMSFT** [comment]
Is there a way to re-queue the build job and tests? The templates do work when I run locally... looks like this particular run hit an issue with the azure backend.

**Comment by bmoore-msft** [comment]
if you add a commit, the pipeline will execute again (I just added one)

**Comment by gimotwanMSFT** [comment]
review comments have been addressed. Would you be able to merge this PR?

**Pull request merged**